### ISSUE TITLE
Stack trace needs more details

### ISSUE THREAD

**Issue by eyurtsev** [langchain-ai/langgraph#2559]
### Privileged issue

- [X] I am a LangChain maintainer, or was asked directly by a LangChain maintainer to create an issue here.

### Issue Content

The RemotException that's surfaced does not have enough information to identify where the error is / which state update fails (InvalidUpdateError)

```python

---------------------------------------------------------------------------
RemoteException                           Traceback (most recent call last)
Cell In[1], line 38
     13 remote_graph = RemoteGraph(graph_name, url=url)
     15 schema = {
     16     "title": "People Name Extractor",
     17     "description": "Extracts names of people from text.",
   (...)
     35     },
     36 }
---> 38 value = remote_graph.invoke(
     39     {
     40         "url": "https://www.apple.com/leadership/",
     41         "json_schema": schema,
     42         "examples": [],
     43     }
     44 )

File ~/.pyenv/versions/3.11.4/envs/graph_3_11_4/lib/python3.11/site-packages/langgraph/pregel/remote.py:772, in RemoteGraph.invoke(self, input, config, interrupt_before, interrupt_after)
    749 def invoke(
    750     self,
    751     input: Union[dict[str, Any], Any],
   (...)
    755     interrupt_after: Optional[Union[All, Sequence[str]]] = None,
    756 ) -> Union[dict[str, Any], Any]:
    757     """Create a run, wait until it finishes and return the final state.
    758 
    759     This method calls `POST [/threads/](http://localhost:8890/threads/){thread_id}[/runs/wait](http://localhost:8890/runs/wait)` if a `thread_id`
   (...)
    770         The output of the graph.
    771     """
--> 772     for chunk in self.stream(
    773         input,
    774         config=config,
    775         interrupt_before=interrupt_before,
    776         interrupt_after=interrupt_after,
    777         stream_mode="values",
    778     ):
    779         pass
    780     try:

File [~/.pyenv/versions/3.11.4/envs/graph_3_11_4/lib/python3.11/site-packages/langgraph/pregel/remote.py:630](http://localhost:8890/home/eugene/.pyenv/versions/3.11.4/envs/graph_3_11_4/lib/python3.11/site-packages/langgraph/pregel/remote.py#line=629), in RemoteGraph.stream(self, input, config, stream_mode, interrupt_before, interrupt_after, subgraphs)
    628         raise GraphInterrupt(chunk.data[INTERRUPT])
    629 elif chunk.event.startswith("error"):
--> 630     raise RemoteException(chunk.data)
    631 # filter for what was actually requested
    632 if mode not in requested:

RemoteException: {'error': 'InvalidUpdateError', 'message': "Expected node t
```

**Comment by anmol-aidora** [comment]
I agree. There should be an option to pass whole exception stacktrace in RemoteException

**Comment by Chinmay1220** [comment]
Hi maintainers, I'd like to work on this issue. I have a patch ready that keeps RemoteException backward-compatible while adding structured context from remote error stream events, including assistant id, stream event, namespace, and payload details. Could you assign this to me so I can open the PR with Fixes #2559? @eyurtsev

**Comment by IgorGanapolsky** [comment]
**Reliability note**

Production automation failures usually reduce to a **contract** issue before a model/prompt issue:

1. **Success criteria** — durable artifact id (message/booking/row), not only node green
2. **Side-effect before complete** — never blind-retry without an idempotency key on the business object
3. **Tool vs path split** — external action can succeed while the node/agent path still errors (silent-green / false-red)

If you have one failing execution + expected vs actual, reply with stack/version and I will send a free root-cause checklist.

**Comment by AaronG18** [comment]
Root cause sits in `libs/langgraph/langgraph/pregel/remote.py`: `RemoteGraph` calls the server `/threads/{id}/runs/wait` (or `/runs/stream`), and when the server raises, the client only surfaces the wrapped `RemoteException` message. The underlying `InvalidUpdateError` - which already knows *which* state key / update failed - is raised server-side and its context is dropped in transit, so the client traceback points at `remote_graph.invoke` with no hint about the offending state update.

Suggested fix (two layers):
- **Server:** when `/runs` fails, include a structured `error.context` in the HTTP error response (node name, the failing state key, the rejected `update`/payload). Highest-leverage change.
- **Client:** `RemoteException` should re-wrap that `error.context` into its message (ideally chaining the original `InvalidUpdateError` type) instead of forwarding only the top-line message.

Lighter client-only improvement: have `InvalidUpdateError.__str__` render the key + offending update, so even the current truncated trace is actionable. Happy to locate the exact serialization spot if maintainers want a PR.

**Comment by AaronG18** [comment]
Opened PR #8845 to fix this (client-only: `RemoteException` now keeps the raw server payload and renders `error` / `message` / `type` / `traceback` instead of flattening to one line). Per the `require-issue-link` bot the PR was auto-closed since external contributors must be assigned - could a maintainer please assign #2559 to me? The bot reopens the PR automatically once assigned.

**Comment by AaronG18** [comment]
Hi maintainers - I'd like to contribute a fix for this issue. Could you please assign it to me (@AaronG18) so the associated PR #8876 can stay open? I've opened it against `main` and am happy to adjust based on your feedback. Thank you!

**Comment by AaronG18** [comment]
Hi @Chinmay1220 - nice, a patch ready. I will leave the implementation to you and just QA. Two things to watch so it stays backward-compatible: (1) RemoteException must remain picklable and keep its current shape for existing consumers; (2) the structured context (assistant id / stream event / namespace / payload) should be populated from the remote error stream events without altering the default message. If you push the branch I can run a before/after diff QA. GL!
